doc/how_to/custom_components/index.md

```diff
@@ -2,6 +2,10 @@
 
```

Comment:
I feel like this "big" topic of custom components and `ReactiveHTML` is very much hidden because its several clicks down

![image](https://github.com/holoviz/panel/assets/42288570/6a9d7364-8b6b-429e-893a-8b57323a97c4)

Also I believe that it would make navigation easier of the sections and cards where aligned as much as possible between `how-to` and `explanation` pages. That would make navigation easier as you don't have to remember seperately where to find ReactiveHTML components material for how-to guides and for explanation.

![image](https://github.com/holoviz/panel/assets/42288570/9d2c678b-6395-4afa-99d8-2fac06acad14)

Reply:
I would recommend restructing the `how-to` navigation by having a `Components` section and moving the following cards there

- construct components
- arrange components
- style components
- build custom components

Reply:
Sounds reasonable to me.